src/base.ts

```diff
@@ -41,76 +30,79 @@ const withPagination = <T>(json: {
   return { content, fieldsStats, pagination, topListing }
 }
 
-export const resolveServiceUrl = (service: Service): string => {
-  let url: string
-  switch (service) {
-    case Service.CAR:
-      url = apiClient.configuration.carServiceUrl
-      break
-    case Service.CATALOGUE:
-      url = apiClient.configuration.catalogueServiceUrl
-      break
-    case Service.SEARCH:
-      url = apiClient.configuration.searchServiceUrl
-      break
-    case Service.DEALER:
-      url = apiClient.configuration.dealerServiceUrl
-      break
-    case Service.OPTION:
-      url = apiClient.configuration.optionServiceUrl
-      break
-    case Service.ANALYTICS:
-      url = apiClient.configuration.analyticsServiceUrl
-      break
-    case Service.USER_NOTIFICATION:
-      url = apiClient.configuration.userNotificationServiceUrl
-      break
-    case Service.TOKEN_REFRESH:
-      url = apiClient.configuration.tokenRefreshServiceUrl
-      break
-    default:
-      throw new Error(`Tried to resolve url of unknown service "${service}"`)
+const authorizationHeader = () => {
+  if (!apiClient.tokens.accessToken) {
+    return null
   }
 
-  if ((url || "").trim()) {
-    return url
-  }
+  return `Bearer ${apiClient.tokens.accessToken}`
+}
 
-  throw new Error(`Missing endpoint configuration for "${service}" service`)
+export interface ApiCallOptions extends Omit<RequestInit, "method" | "body"> {
+  recaptchaToken?: string
+  headers?: Record<string, string>
+  host?: string
 }
 
-const authorizationHeader = () => {
-  if (!apiClient.tokens.accessToken) {
-    return null
+type Method = "GET" | "POST" | "PUT" | "PATCH" | "DELETE"
+
+interface RequestOptions extends ApiCallOptions {
+  method?: Method
+}
+
+export const getHost = (host: string = null) => {
+  if (!apiClient.configuration.host) {
+    throw new Error(
+      'ApiClient not configured, please run: ApiClient.configure({ apiGatewayUrl: "your.api.gateway" }'
+    )
   }
 
-  return `Bearer ${apiClient.tokens.accessToken}`
+  return host || apiClient.configuration.host
 }
 
-export const fetchPath = async (
-  service: Service,
-  path: string,
-  options = {}
-) => {
-  const prefix = resolveServiceUrl(service)
-  const url = `${prefix}/${stripLeadingSlash(path)}`
+const buildHeaders = ({
+  headers = {},
+  recaptchaToken,
+}: RequestOptions): Record<string, string> => {
+  const auth = authorizationHeader()
+
+  return {
+    "Content-Type": "application/json",
+    Accept: `application/vnd.carforyou.${apiClient.version}+json`,
+    ...(auth ? { Authorization: auth } : {}),
+    ...(recaptchaToken ? { "Recaptcha-Token": recaptchaToken } : {}),
+    ...headers,
+  }
+}
+
+export const fetchPath = async ({
+  path,
+  body,
+  options,
+}: {
+  path: string
+  body?: string
+  options: RequestOptions
+}) => {
+  const {
+    method = "GET",
+    host = null,
+    headers,
+    recaptchaToken,
+    ...fetchOptions
```

Comment:
I checked this in my PR and we do not use it in the code as far as I can tell so why not just remove it and be explicit if we need a specific feature in the future and add the used feature instead of "whitelisting" a lot of the default interface

Reply:
I recall using it a couple of times when debugging. It's typed now so I'm not that much concerned here.